src/main/java/org/elasticsearch/cluster/action/index/NodeIndexDeletedAction.java

```diff
@@ -64,32 +75,56 @@ public void remove(Listener listener) {
     }
 
     public void nodeIndexDeleted(final ClusterState clusterState, final String index, final String nodeId) throws ElasticsearchException {
-        DiscoveryNodes nodes = clusterState.nodes();
+        final DiscoveryNodes nodes = clusterState.nodes();
         if (nodes.localNodeMaster()) {
-            threadPool.generic().execute(new Runnable() {
+            threadPool.generic().execute(new AbstractRunnable() {
+
+                @Override
+                public void onFailure(Throwable t) {
+                    logger.warn("failed to ack index store deleted for  index {}", t, index);
```

Comment:
extra space between 'for' and 'index'. Also, we typically log the index name first under []: `[{}] failed to ack index store deleted' . If I read things correctly, this also logged when something went wrong in innerNodeIndexDeleted so maybe change to "fail to ack index & store deletion"